src/GslCore/AstTypes.fs

```diff
@@ -461,8 +543,13 @@ let posFromList (nodes: AstNode list) : SourcePosition option =
 /// Create a parse error with message and position.
 let createParseError msg pos = ParseError({x=msg; pos=pos})
 
+let posOptionToList (pos:SourcePosition option) =
+    match pos with
+    | Some p -> [p]
+    | None -> []
+
 /// Wrap a value with position taken from another node.
-let nodeWrapWithNodePosition (node: AstNode) v = {x = v; pos = node.pos}
+let nodeWrapWithNodePosition (node: AstNode) v = {x = v; pos = node.pos |> posOptionToList}

```

Comment:
Good point.  Was hoping this would fix the problem with multiply recursive function expansion losing the middle positions but did not,.  Implemented anyway,